packages/medusa/src/utils/authenticate-middleware.ts

```diff
@@ -25,34 +33,41 @@ export const authenticate = (
     next: NextFunction
   ): Promise<void> => {
     const authTypes = Array.isArray(authType) ? authType : [authType]
-    const authModule = req.scope.resolve<IAuthModuleService>(
-      ModuleRegistrationName.AUTH
-    )
 
     // @ts-ignore
     const session: MedusaSession = req.session || {}
 
     let authUser: AuthUserDTO | null = null
     if (authTypes.includes(SESSION_AUTH)) {
-      if (session.auth_user && session.scope === authScope) {
+      if (
+        session.auth_user &&
+        isValidScope(authScope, session.auth_user.scope)
+      ) {
         authUser = session.auth_user
       }
     }
 
-    if (!authUser && authTypes.includes(BEARER_AUTH)) {
+    if (authTypes.includes(BEARER_AUTH)) {
       const authHeader = req.headers.authorization
+
       if (authHeader) {
         const re = /(\S+)\s+(\S+)/
         const matches = authHeader.match(re)
 
-        // TODO: figure out how to obtain token (and store correct data in token)
         if (matches) {
           const tokenType = matches[1]
           const token = matches[2]
-          if (tokenType.toLowerCase() === "bearer") {
-            authUser = await authModule
-              .retrieveAuthUserFromJwtToken(token, authScope)
-              .catch(() => null)
+          if (tokenType.toLowerCase() === BEARER_AUTH) {
+            // get config jwt secret
+            // verify token and set authUser
+            const { jwt_secret } =
+              req.scope.resolve("configModule").projectConfig
+
+            const verified = jwt.verify(token, jwt_secret) as JwtPayload
+
```

Comment:
thought: maybe this should move to a util.

Reply:
Added `stringEqualsOrRegexMatch` util 😄

Reply:
Added a util for string equality/regex matching 😄